### PR TITLE
Align file configuration with latest changes to spec

### PR DESCRIPTION
**Pull request by jack-berg**
Notes:

- Change env substitution syntax from ${env:MY_ENV_VAR} to ${MY_ENV_VAR}
- Rename and combine ConfigurationReader / ConfigurationFactory to FileConfiguration. 
- Change "interpet" to "create", break out dedicate "parse", and "create" methods in addition to the convenience method "parseAndCreate".
- Document usage in README

### CODE REVIEW

**Comment by brunobat** [comment]
@jack-berg, Will we still be able to use all functionality of the autoconfiguration without this file based configuration?

**Comment by jack-berg** [comment]
> @jack-berg, Will we still be able to use all functionality of the Autoconfiguration without this file based configuration?
Do we have somewhere an explanation of how the Autoconfiguration will look like after this file config is completed?

@brunobat we are committed to backwards compatibility, so autoconfiguration will always need to continue to work without file based configuration.

In terms of how Autoconfiguration will look after file config is completed, a lot of the reference material for that will be the spec itself. There are things that are unique to java, like the ability to customize configuration via various SPI hooks. Hard to say exactly how all the mechanisms will work with file configuration. I think some of the SPIs only make sense to fill shortcomings of the environment variable scheme, but others make sense to reuse. 

I've been struggling to get engagement from the Java SIG so a lot open questions remain.

**Comment by jack-berg** [comment]
@open-telemetry/java-approvers PTAL if possible. The changes in this PR don't introduce anything new - just changes to align with the spec. Would really like to get this merged so I can work against code / vocabulary that reflects the current state of things and avoid nasty rebases.

File: sdk-extensions/incubator/README.md [hunk 1]
```diff
@@ -2,6 +2,26 @@
...
+OpenTelemetrySdk openTelemetrySdk;
+try (FileInputStream yamlConfigFileInputStream = new FileInputStream("/path/to/config.yaml")) {
+  openTelemetrySdk = FileConfiguration.parseAndCreate(yamlConfigFileInputStream);
+}
+// ...proceed with application after successful initialization of OpenTelemetrySdk
+```
+
+Notes:
+* Environment variable substitution is supported as [defined in the spec](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/configuration/file-configuration.md#environment-variable-substitution)
+* Currently, there is no support for the SPIs defined in  [opentelemetry-sdk-extension-autoconfigure-spi](../autoconfigure-spi). Only built in samplers, processors, exporters, etc can be configured.
```
Comment (trask) [inline sdk-extensions/incubator/README.md:22]:
```suggestion
* Currently, there is no support for the SPIs defined in [opentelemetry-sdk-extension-autoconfigure-spi](../autoconfigure-spi). Only built in samplers, processors, exporters, etc can be configured.
```

**Review by trask — APPROVED** [review]

**Pull request merged**